lib/bolt/transport/local/connection.rb

```diff
@@ -44,7 +45,16 @@ def copy_file(source, dest)
         end
 
         def execute(command)
-          Open3.popen3(command)
+          if Bolt::Util.windows?
+            command += "\r\nif (!$?) { if($LASTEXITCODE) { exit $LASTEXITCODE } else { exit 1 } }"
+            script_file = Tempfile.new(['wrapper', '.ps1'])
```

Comment:
Should this use the `Powershell::Tmpdir` (or tempdir methods if they stay in Powershell)? It seems weird to make an exceptional tempfile not in the configured tempdir...

Reply:
You're right, this should be in the tmpdir. The call in `copyfile` is okay I think, since that's a "local" tempfile, not a "remote" tempfile.